lib/new_inject.rb

```diff
@@ -0,0 +1,109 @@
+# class Array
+#
+#   def new_inject *arg, &block
+#     if arg == []
+#       copy = self.dup
+#       result = copy.delete_at(0)
+#       copy.each do |element|
+#         result = block.call(result, element)
+#       end
+#       result
+#     else
+#       result = arg[0]
+#       self.each do |element|
+#         result = block.call(result, element)
+#       end
+#       result
+#     end
+#   end
+#
+#   def recursion_inject *arg, &block
+#     if arg == []
+#       self_copy = self.dup
+#       result = self_copy.delete_at(0)
+#       element = self_copy.delete_at(0)
+#       result = block.call(result, element)
+#       if self_copy == []
+#         return result
+#       else
+#         return self_copy.recursion_inject result, &block
+#       end
+#     else
+#       self_copy = self.dup
+#       result = arg[0]
+#       element = self_copy.delete_at(0)
+#       result = block.call(result, element)
+#       if self_copy == []
+#         return result
+#       else
+#         return self_copy.recursion_inject result, &block
+#       end
+#     end
+#   end
+#
+# end
+
+
+
+class Array
+
+  def new_inject_with_arg arg, &block
+    result = arg
+    self.each do |element|
+      result = block.call(result, element)
+    end
+    result
+  end
+
+  def new_inject_without_arg &block
+    copy = self.dup
+    result = copy.delete_at(0)
+    copy.each do |element|
+      result = block.call(result, element)
+    end
+    result
+  end
+
+  def new_inject *arg, &block
+    if arg == []
+      return self.new_inject_without_arg &block
+    else
+      return self.new_inject_with_arg arg[0], &block
+    end
+  end
+
+  def recursion_inject_with_arg arg, &block
+    self_copy = self.dup
+    result = arg
+    element = self_copy.delete_at(0)
+    result = block.call(result, element)
+    if self_copy == []
+      return result
+    else
+      return self_copy.recursion_inject_with_arg result, &block
+    end
+  end
+
+  def recursion_inject_without_arg &block
+    self_copy = self.dup
+    result = self_copy.delete_at(0)
+    element = self_copy.delete_at(0)
+    puts "result - #{result} element - #{element}"
+    result = block.call(result, element)
+    if self_copy == []
+      puts "result - #{result}"
+      return result
+    else
+      return self_copy.recursion_inject_without_arg &block
```

Comment:
Lint/AmbiguousOperator: Ambiguous block operator. Parenthesize the method arguments if it's surely a block operator, or add a whitespace to the right of the `&` if it should be a binary AND.